graylog2-web-interface/src/components/indices/IndexSetFieldTypesList.tsx

```diff
@@ -0,0 +1,204 @@
+  const [editingField, setEditingField] = useState<IndexSetFieldType | null>(null);
+
+  const handleOnClose = useCallback(() => {
+    setEditingField(null);
+  }, []);
+
+  const handleOnOpen = useCallback((fieldType: IndexSetFieldType) => {
+    setEditingField(fieldType);
+  }, []);
+
+  const initialSelection = useMemo(() => [indexSetId], [indexSetId]);
+  const [query, setQuery] = useState('');
+  const [activePage, setActivePage] = useState(1);
+  const { data: { fieldTypes }, isLoading: isOptionsLoading } = useFiledTypes();
+  const { layoutConfig, isInitialLoading: isLoadingLayoutPreferences } = useTableLayout({
+    entityTableId: ENTITY_TABLE_ID,
+    defaultPageSize: DEFAULT_LAYOUT.pageSize,
+    defaultDisplayedAttributes: DEFAULT_LAYOUT.displayedColumns,
+    defaultSort: DEFAULT_LAYOUT.sort,
+  });
+  const searchParams = useMemo(() => ({
+    query,
+    page: activePage,
+    pageSize: layoutConfig.pageSize,
+    sort: layoutConfig.sort,
+  }), [activePage, layoutConfig.pageSize, layoutConfig.sort, query]);
+  const { mutate: updateTableLayout } = useUpdateUserLayoutPreferences(ENTITY_TABLE_ID);
+  const onPageChange = useCallback(
+    (newPage: number, newPageSize: number) => {
+      if (newPage) {
+        setActivePage(newPage);
+      }
+
+      if (newPageSize) {
+        updateTableLayout({ perPage: newPageSize });
+      }
+    }, [updateTableLayout],
+  );
+
+  const onPageSizeChange = useCallback((newPageSize: number) => {
+    setActivePage(1);
+    updateTableLayout({ perPage: newPageSize });
+  }, [updateTableLayout]);
+
+  const onSortChange = useCallback((newSort: Sort) => {
+    setActivePage(1);
+    updateTableLayout({ sort: newSort });
+  }, [updateTableLayout]);
+
+  const onSearch = useCallback((newQuery: string) => {
+    setActivePage(1);
+    setQuery(newQuery);
+  }, []);
+
+  const onResetSearch = useCallback(() => onSearch(''), [onSearch]);
+
+  const onColumnsChange = useCallback((displayedAttributes: Array<string>) => {
+    updateTableLayout({ displayedAttributes });
+  }, [updateTableLayout]);
+  const { isLoading, data: { list, pagination, attributes }, refetch } = useIndexSetFieldTypes(indexSetId, searchParams, { enabled: !isLoadingLayoutPreferences });
+
+  const customColumnRenderers = useMemo(() => ({
+    attributes: {
+      type: {
+        renderCell: (item: string) => <span>{fieldTypes[item]}</span>,
+      },
+      is_custom: {
+        renderHeader: () => null,
+        renderCell: (isCustom: boolean) => (isCustom ? <Label bsStyle="primary">custom</Label> : null),
+      },
+    },
+  }), [fieldTypes]);
+
+  const openEditModal = useCallback((fieldType: IndexSetFieldType) => {
+    handleOnOpen(fieldType);
+  }, [handleOnOpen]);
+
+  const renderActions = useCallback((fieldType: IndexSetFieldType) => (
+    <>
+      <Button onClick={() => openEditModal(fieldType)}
+              role="button"
+              bsSize="xsmall"
+              disabled={fieldType.isReserved}
```

Comment:
The help icon doesn't look nice. I will add the reserved column and change the edit button title (the one which appears on button hover)

Reply:
I had the idea, because that is what we use in other cases as well and it would be nice to unify the UI.
<img width="177" alt="image" src="https://github.com/Graylog2/graylog2-server/assets/46300478/0529251e-4d6c-4429-9ffe-58059a7cb478">

<img width="325" alt="image" src="https://github.com/Graylog2/graylog2-server/assets/46300478/6e030967-a562-423a-a4b6-9fb42e6bbaf5">


Having a "Is reserved" column is also nice, but I think we should explain somewhere what "reserved" means in this context, e.g. in the page header.